private/pkg/storage/multi.go

```diff
@@ -76,6 +77,20 @@ func newMultiReadBucket(
 }
 
 func (m *multiReadBucket) Get(ctx context.Context, path string) (ReadObjectCloser, error) {
+	if m.overlay {
+		// If overlay is enabled, attempt a Get operation against the first bucket. If the file is
+		// found, we avoid the potentially expensive Stat call.
+		readObjectCloser, err := m.delegates[0].Get(ctx, path)
+		if err != nil {
+			if !errors.Is(err, fs.ErrNotExist) {
+				return nil, err
+			}
```

Comment:
Probably a comment here "fallthrough to expensive logic to check every delegate"

Reply:
Added in https://github.com/bufbuild/buf/pull/3393/commits/386db91e616773a90fc8fb7a1129142cbea3c43d